src/pyhf/contrib/extended_modifiers/purefunc.py

```
@@ -0,0 +1,233 @@
+
+        def __init__(self, pdfconfig, transforms):
+            self.config = pdfconfig
+            self.transforms = transforms
+            self.required_parsets = {}
+            self.builder_data = {"local": {}, "global": {"symbols": set()}}
+            self.parsed_expressions = {}
+            self.languages = {}
+            self.free_symbols = set()
+            self.parse_expressions()
+
+        def parse_expressions(self):
```

Ruff (0.15.21)

[warning] Too many branches (13 > 12)

(PLR0912)